tests/protocol/test_hostplugin.py

```diff
@@ -564,151 +562,153 @@ def test_validate_get_extension_artifacts(self):
                     self.assertTrue(k in actual_headers)
                     self.assertEqual(expected_headers[k], actual_headers[k])
 
-    @patch("azurelinuxagent.common.utils.restutil.http_get")
-    def test_health(self, patch_http_get):
+    def test_health(self):
         host_plugin = self._init_host()
 
-        patch_http_get.return_value = MockResponse('', 200)
-        result = host_plugin.get_health()
-        self.assertEqual(1, patch_http_get.call_count)
-        self.assertTrue(result)
-
-        patch_http_get.return_value = MockResponse('', 500)
-        result = host_plugin.get_health()
-        self.assertFalse(result)
-
-        patch_http_get.side_effect = IOError('client IO error')
-        try:
-            host_plugin.get_health()
-            self.fail('IO error expected to be raised')
-        except IOError:
-            # expected
-            pass
-
-    @patch("azurelinuxagent.common.utils.restutil.http_get",
-           return_value=MockResponse(status_code=200, body=b''))
-    @patch("azurelinuxagent.common.protocol.healthservice.HealthService.report_host_plugin_versions")
-    def test_ensure_health_service_called(self, patch_http_get, patch_report_versions):
+        with patch("azurelinuxagent.common.utils.restutil.http_get") as patch_http_get:
```

Comment:
I moved these patch decorators to be statements while I was debugging the issue with the mock request. Although this change is no longer needed I'm leaving there should we need to debug more.